mautic/mautic
# Autowire models.

| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ x ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 



#### Description:
Provides autowired models and allows to use `ModelFactory->getModel(ModelClass::class)`.
As a side feature adds ability to mark a class as autowired in old `config.php` config file.


#### Steps to test this PR:
This is another broad change of the system, so please test several pages. 

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check the website is working.
